### ISSUE TITLE
More Styling for WorldMap

### ISSUE THREAD

**Issue by ahappypie** [grommet/grommet#1337]
WorldMap fit really well in a project of mine, but when I went to customize it, found the styling options lacking.
### Expected Behavior

Ideally, both the static colorIndex and mouseOver color should be customizable, and mouseOver color should be independent of any onClick behavior.
### Actual Behavior

mouseOver color is linked to static colorIndex and onClick is required to highlight a continent. Furthermore, onClick behavior changes the styling of the cursor to a hand.
### URL, screen shot, or Codepen exhibiting the issue
![original](https://cloud.githubusercontent.com/assets/2286880/25679019/b8190146-3000-11e7-8f2b-09db481dd8e3.gif)

### Steps to Reproduce
 1. Default functionality of WorldMap at the [docs](https://grommet.github.io/docs/world-map/examples)

### Solution
I am happy to submit a pull request with my solution, which adds both individual highlighting of continents:
![highlightonly](https://cloud.githubusercontent.com/assets/2286880/25679170/2935f744-3001-11e7-8c0c-682e0cf041f0.gif)
And highlighting with support for onClick:
![withclick](https://cloud.githubusercontent.com/assets/2286880/25679193/41ecf8b4-3001-11e7-905a-ee88a80a6622.gif)
I modified grommet/components/WorldMap.js for the new functionality and added one class to grommet/scss/grommet-core/_objects.world-map.scss to deal with the cursor.
This is how I constructed the WorldMap above (though value isn't actually used here):
```jsx
const SERIES = [
  {continent: 'NorthAmerica', label: 'North America', value: 40,
    colorIndex: 'graph-1', highlightColorIndex: 'graph-4', hover: true},
  {continent: 'SouthAmerica', label: 'South America', value: 30,
    colorIndex: 'accent-2', highlightColorIndex: 'graph-3', hover: true},
  {continent: 'Europe', label: 'Europe', value: 20,
    colorIndex: 'unset', highlightColorIndex: 'graph-2', hover: true},
  {continent: 'Africa', label: 'Africa', value: 10,
    colorIndex: 'graph-2', highlightColorIndex: 'critical', onClick: () => {
      alert('clicking also works!');
    }},
  {continent: 'Asia', label: 'Asia', value: 15,
    colorIndex: 'graph-3', highlightColorIndex: 'accent-2', hover: true},
  {continent: 'Australia', label: 'Australia', value: 10,
    colorIndex: 'graph-4', highlightColorIndex: 'graph-1', hover: true}
];
<WorldMap series={SERIES} />
```

### Your Environment

Developed under:
* Grommet version: 1.4.0
* Browser Name and version: Chrome 57
* Operating System and version (desktop or mobile): MacOS 10.10.5

**Comment by dgennetten** [comment]
@ahappypie, we welcome pull requests! Go ahead and submit.

**Comment by ericsoderberghp** [comment]
I would like to understand a bit more on when it would make sense to change the color on hover. For instance, the example shows changing the color of South America to a completely different color that matches Asia. That doesn't seem the like a consistent usage of color.

I agree with needing the enhancement to have hover and click be separably available.

I am about to submit some other changes to WorldMap to allow indicating individual places. If I don't hear from you in the next day or two, I will merge over the hover|click behavior separation.

**Comment by karatechops** [comment]
I've closed #1369 and documented the feedback in 2.x's Map component (#1473).
